sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -457,28 +460,35 @@ public String getId() {
      */
     @Beta(value = Beta.SinceVersion.V4_14_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
     public Mono<Void> openConnectionsAndInitCaches() {
-        int retryCount = Configs.getOpenConnectionsRetriesCount();
-
         if(isInitialized.compareAndSet(false, true)) {
-            return this.getFeedRanges().flatMap(feedRanges -> {
-                List<Flux<FeedResponse<ObjectNode>>> fluxList = new ArrayList<>();
-                SqlQuerySpec querySpec = new SqlQuerySpec();
-                querySpec.setQueryText("select * from c where c.id = @id");
-                querySpec.setParameters(Collections.singletonList(new SqlParameter("@id",
-                    UUID.randomUUID().toString())));
-
-                for (int i = 0; i < retryCount; i++) {
-                    for (FeedRange feedRange : feedRanges) {
-                        CosmosQueryRequestOptions options = new CosmosQueryRequestOptions();
-                        options.setFeedRange(feedRange);
-                        CosmosPagedFlux<ObjectNode> cosmosPagedFlux = this.queryItems(querySpec, options,
-                            ObjectNode.class);
-                        fluxList.add(cosmosPagedFlux.byPage());
-                    }
+            return this.getPartitionKeyRanges().flatMap(partitionKeyRanges -> {
+                List<Mono<CosmosItemResponse<ObjectNode>>> monoList = new ArrayList<>();
+                for(PartitionKeyRange partitionKeyRange : partitionKeyRanges) {
+                    CosmosItemRequestOptions requestOptions = new CosmosItemRequestOptions();
+                    cosmosItemRequestOptionsAccessor.setPartitionKeyRangeId(requestOptions, partitionKeyRange.getId());
```

Comment:
In general I don't think we have way to suppress logging on service via client, and we shouldn't . On noise , agree during the start it will log few 410/1002, but not afterward during the actual workload, so we should be fine

Reply:
Also brain-stormed with @ealsur - here is a counter proposal:
Insteado f using fixed partition key value use getFeedranges instead of getPkRangesIds - that way you get a feed range for each physical partition. Then you can use the feedrange's minRange (always inclusive) as the partition key - basically you know an epk/logical partition on each physical partition. That way you would always get a 404 (if you get a 410 it is due to real split and you can just let the normal retry flow happen). Basically this allows removing all the 410 related special-casing and it also reduces the noise of "suspicious" status codes (410/xxx) during client initialization.

Reply:
Server contract need the actual value of partition key in header, how we will convert the range to actual value

Reply:
Alternative: Use getPKRangesIds -> Send 1 Change Feed request with the PKRangeId header to each replica with maxitemcount 1. That should be cheap and you don't need any special change in the HA stack?